loader/loader_test.go

```diff
@@ -3329,3 +3329,27 @@ secrets:
 			Content:     "Shadoks",
 		}})
 }
+
+func TestLoadDeviceMapping(t *testing.T) {
+	config, err := loadYAML(`
+name: load-device-mapping
+services:
+  test:
+    devices:
```

Comment:
Could you also introduce the test case for the object/map case (with an extension)?